tools/scylla-nodetool.cc

```diff
@@ -2644,6 +2645,241 @@ void table_stats_operation(scylla_rest_client& client, const bpo::variables_map&
+}
+
+void tasks_list_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (!vm.contains("module")) {
+        throw std::invalid_argument("required parameter is missing: module");
+    }
+    auto module = vm["module"].as<sstring>();
+    std::unordered_map<sstring, sstring> params;
+    if (vm.contains("keyspace")) {
+        params["keyspace"] = vm["keyspace"].as<sstring>();
+    }
+    if (vm.contains("table")) {
+        params["table"] = vm["table"].as<sstring>();
+    }
+    if (vm.contains("internal")) {
+        params["internal"] = "true";
+    }
+    int period = vm["period"].as<int>();
+    int iterations = vm["iterations"].as<int>();
+
+    auto res = client.get(format("/task_manager/list_module_tasks/{}", module), params);
+    print_stats_list(res);
+
+    for (auto i = 0; i < iterations; ++i) {
+        sleep(period);
+        auto res = client.get(format("/task_manager/list_module_tasks/{}", module), params);
+        fmt::print("\n\n");
+        print_stats_list(res);
+    }
+}
+
+void tasks_modules_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    auto res = client.get("/task_manager/list_modules");
+
+    for (auto& x: res.GetArray()) {
+        fmt::print("{}\n", rjson::to_string_view(x));
+    }
+}
+
+void tasks_set_ttl_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (!vm.contains("ttl")) {
+        throw std::invalid_argument("required parameter is missing: ttl");
+    }
+    auto new_ttl = vm["ttl"].as<uint32_t>();
+    std::unordered_map<sstring, sstring> params = {{ "ttl", fmt::format("{}", new_ttl) }};
+
+    auto res = client.post("/task_manager/ttl", std::move(params));
+
+    fmt::print("Set ttl value to {}\n", new_ttl);
+}
+
+void tasks_status_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (!vm.contains("id")) {
+        throw std::invalid_argument("required parameter is missing: id");
+    }
+    auto id = vm["id"].as<sstring>();
+
+    auto res = client.get(format("/task_manager/task_status/{}", id));
+
+    print_status(res);
+}
+
+void tasks_tree_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    std::vector<rjson::value> res;
+    if (vm.contains("id")) {
+        auto id = vm["id"].as<sstring>();
+
+        res.push_back(client.get(format("/task_manager/task_status_recursive/{}", id)));
+
+        print_trees(res);
+        return;
+    }
+
+    auto module_res = client.get("/task_manager/list_modules");
+    for (const auto& module : module_res.GetArray()) {
+        auto list_res = client.get(format("/task_manager/list_module_tasks/{}", module.GetString()));
+        for (const auto& stats : list_res.GetArray()) {
+            try {
+                res.push_back(client.get(format("/task_manager/task_status_recursive/{}", stats.GetObject()["task_id"].GetString())));
+            } catch (const api_request_failed&) {
+                // Task has already been unregistered.
```

Comment:
Need to match the status code and only ignore the specific status code which means we attempted to access unregistered task.